scio-core/src/main/scala/com/spotify/scio/io/ScioIO.scala

```diff
@@ -24,6 +24,33 @@ import com.spotify.scio.values.SCollection
 
 import scala.concurrent.Future
 
+sealed trait TapT[A] {
```

Comment:
@jto I think we could move this to `Tap.scala`?

Reply:
No. The only use of the class is to encode the fact the write and tap can either return a `Tap[A]` or a `Tap[Nothing]`. It's actually not strictly related to `Tap` but I can't find a better name...

Reply:
yeah on a second thought I agree with you! 👍